bittensor/_dataset/dataset_impl.py

```diff
@@ -1,536 +1,490 @@
-        r""" Get the size (in byte) of a folder inside the data_dir.
+            if batch_count > self.num_batches:
+                break
+            raw_text = self.async_run(self.get_text(text_hash=text_hash), loop=loop)
+
+            if not queue.full():
+            # skip queue if it is full
+
+                queue.put(raw_text)
+
+    def build_datasets(self, datasets:List[str]=None, save:bool=None, load:bool=None, loop:'asyncio.loop'=None):
+        """ Building all of the datasets specified by getting each of their 
+            text hashes from IPFS or local
         Args:
-            folder (str):
-                The name of the folder
-        
-        Returns:
-            total_size (int):
-                The memory size of the folder (in byte). 
+            datasets (List[str]):
+                Axon to serve.s
+            save (bool):
+                Save the dataset hashes locally.
+            load (bool):
+                Load the dataset hashes locally
+            loop (asyncio.Loop):
+                Asyncio loop 
         """
-        total_size = 0
-        full_path = os.path.expanduser(os.path.join(self.data_dir, folder))
-        for dirpath, dirnames, filenames in os.walk(full_path):
-            for f in filenames:
-                fp = os.path.join(dirpath, f)
-                # skip if it is symbolic link
-                if not os.path.islink(fp):
-                    total_size += os.path.getsize(fp)
-
-        return total_size
-
-    def load_hash(self, file_meta):
-        r""" Load a hash from disk.
-        Args:
-            file_meta (dict of str: int):
-                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
+        if datasets == None:
+            datasets = self.datasets
+        if save == None:
+            save = self.save_dataset
+        if load == None:
+            load = self.load_dataset
 
-        Returns:
-            text (str): 
-                The text in the file.                
-        """
+        self.dataset_size = 0
 
-        full_path = os.path.expanduser(os.path.join(self.data_dir, file_meta['Folder'], file_meta['Hash']))
-        if os.path.exists(full_path):
-            try:
-                with open(full_path, mode='r') as f:
-                    text = f.read()
+        all_text_file_metas = []
+        dataset_hash_map = {}
 
-                logger.success("Loaded from disk:".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-            except Exception:
-                logger.success("Could not load from disk:".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-                pass
+        if len(dataset_hash_map) == 0:
+            tasks = []
 
-            return text
-        
-        return None
+            # Gather dataset hashes async as their state is independent.
+            for dataset in datasets:
+                tasks += [self.build_dataset(dataset=dataset, save=save, load=load, loop=loop)]
```

Comment:
I think `self.build_dataset` should have a more descriptive name like `build_single_dataset`. It's too similar to this functions name (one character off) and I misunderstood why we were recursively calling this function in an asyncio loop lol